### PR TITLE
Chat improvements (#2288)

### PR DESCRIPTION
**Pull request by Frago9876543210**
## Introduction
Fix for #2288 and some improvements about mute

### API changes
Added:
`Player::setMuted()`
`Player::isMuted()`

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -593,6 +604,7 @@ public function setOp(bool $value){
 		}else{
 			$this->server->removeOp($this->getName());
 		}
+		$this->setRemoveFormat(!$value);
```
Comment (dktapps) [inline src/pocketmine/Player.php]: This can be overridden by plugins, which is why I didn't apply such a fix myself already. Additionally this is not related to mute.

**Comment by Frago9876543210** [comment]
@dktapps You offers to do handle messages as parts? For example, for processing mute, color messages, etc.
